src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -204,28 +220,47 @@ private void StartNewLoggingThread()
                 Name = _name + " Thread"
             };
             _loggingThread.Start();
+            if (compress)
+            {
+                _compressLoggingThread = new Thread(BackgroundLoggerCompress)
+                {
+                    IsBackground = true,
+                    Name = _name + "Compress Thread"
+                };
+                _compressLoggingThread.Start();
+            }
+            else
+            {
+                _compressLoggingThread = null;
+            }
         }
 
         public void EndLogging()
         {
             _keepLogging.Lower();
             _hasEntries.Set();
+            _readyToCompress.Set();
+            _tokenSource.Cancel();
             _loggingThread.Join(TimeToWaitForLoggingToEndInMilliseconds);
-
+            _compressLoggingThread.Join();
         }
 
         private FileStream GetNewStream(long maxFileSize)
         {
+            var logFiles = Directory.GetFiles(_path, $"{_dateString}.*.log");
```

Comment:
Should I consider a network error?

Reply:
What do you mean, when we are failing to get the directory is on the network?

Reply:
yes, when the path to the log directory is on a remote computer.
This is the only exception I can think we can suddenly get from `GetFIiles` after we already use it
(accept to when someone deleting the directory and create a file with the same name)

Reply:
There are a lot of things that this can be.
Out of file descriptors, I/O errors, permissions issues, etc.